Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -124,7 +124,15 @@ function getUpdateHistoryRequest(webAppPublishKuduUrl: string, isDeploymentSucce
 
     var collectionUrl = tl.getVariable('system.TeamFoundationCollectionUri'); 
     var teamProject = tl.getVariable('system.teamProject');
-
+	
+	var type = "Deployment";
+	var commitId = tl.getVariable('build.sourceVersion');
+	var repoName = tl.getVariable('build.repository.name');
```

Comment:
Check does tl.getVariable has an option to specify is it mandatory or not. In that sense you don't need to handle each input instead your task execution stops when task lib does not find this variable.